src/linter.ts

```diff
@@ -131,19 +131,40 @@ class Linter {
             throw new Error(`formatter '${formatterName}' not found`);
         }
 
-        const output = formatter.format(this.failures, this.fixes);
+        // Sort failures by failure's line and character.
+        const failures = this.sortFailures(this.failures);
+        const fixes = this.sortFailures(this.fixes);
```

Comment:
the fixes are actually never really used by a formatter, so they don't need to be sorted.